server/app/views/components/FieldWithLabel.java

```diff
@@ -223,7 +224,20 @@ public FieldWithLabel setScreenReaderText(String screenReaderText) {
   public FieldWithLabel setFieldErrors(
       Messages messages, ImmutableSet<ValidationErrorMessage> errors) {
     this.messages = messages;
-    this.fieldErrors = errors;
+    this.fieldErrors =
+        errors.stream()
+            .map(
+                (ValidationErrorMessage vem) ->
+                    new ValidationError(vem.key().getKeyName(), vem.key().getKeyName(), vem.args()))
+            .collect(ImmutableSet.toImmutableSet());
+
+    return this;
```

Comment:
Not-blocking: Nit. Since this method overload will hopefully remain temporary, what about having it call the more preferred method with the converted errors.

e.g.

public FieldWithLabel setFieldErrors(Messages messages, ImmutableSet<ValidationErrorMessage> errors) {
  return setFieldErrors(messages, errors.stream().....);
}

Reply:
Unfortunately `FieldWithLabel setFieldErrors(Messages messages, ImmutableSet<ValidationError> errors)` and `setFieldErrors(Messages messages, ImmutableSet<ValidationErrorMessage> errors)` have the same erasure so we can't do that overload. It is temporary though since we'll do the broader refactor after this PR goes in.